src/api/tokensApi.js

```diff
@@ -8,28 +8,86 @@
 import requestExplorerServiceV1 from './axiosInstance';
 import { get } from 'lodash';
 
+const handleResponse = (response) => {
+  // If status is not retrieved, we assume an internal error ocurred, giving the status code 500
+  // Currently 200 is always returned for success responses
+  if (get(response, 'status', 500) !== 200) {
+    return {
+      'error': true,
+    };
+  }
+
+  return {
+    ...response,
+    error: false,
+  };
+};
+
 const tokensApi = {
-    async getList(searchText, sortBy, order, searchAfter) {
-        const data = {
-            "search_text": searchText,
-            "sort_by": sortBy,
-            "order": order,
-            "search_after": searchAfter.join(",")
-        };
-
-        let result = await requestExplorerServiceV1.get(`tokens`, { params: data });
-
-        // If status is not retrieved, we assume an internal error ocurred, giving the status code 500
-        // Currently 200 is always returned for success responses
-        if (get(result, 'status', 500) !== 200) {
-            return {
-                "error": true
-            }
-        }
-
-        result.error = false;
-        return result;
-    },
+  /**
+   * Downloads a list of tokens that match the searched text
+   *
+   * @param {String} searchText The text to match against the token values
+   * @param {String} sortBy field to sort by, possible values are (name, symbol, uid)
+   * @param {String} order Order to sort the fields, possible values are ('desc', 'asc')
+   * @param {Array[String]} searchAfter Search after this result, this should use the sort attribute of
+   * the returned hit
+   *
+   * @return {Promise}
+   */
+  async getList(searchText, sortBy, order, searchAfter) {
+    const data = {
+      'search_text': searchText,
+      'sort_by': sortBy,
+      'order': order,
+      'search_after': searchAfter.join(','),
+    };
+
+    const response = await requestExplorerServiceV1.get('tokens', { params: data });
+
+    return handleResponse(response);
+  },
+
+  /**
+   * Downloads a list of token balances given an token_id
+   *
+   * @param {String} tokenId Token id to filter token address balances
+   * @param {String} sortBy field to sort by, possible values are (total, unlocked_balance, locked_balance)
+   * @param {String} order Order to sort the fields, possible values are ('desc', 'asc')
+   * @param {Array[String]} searchAfter Search after this result, this should use the sort attribute of
+   * the returned hit
+   *
+   * @return {Promise}
+   */
+  async getBalances(tokenId, sortBy, order, searchAfter) {
+    const data = {
+      'token_id': tokenId,
+      'sort_by': sortBy,
+      'order': order,
+      'search_after': searchAfter.join(',')
+    };
+
+    const response = await requestExplorerServiceV1.get('token_balances', { params: data });
```

Comment:
Should we set a timeout in the requests? It seems Axios' default is to have no timeout at all.

Reply:
Axios changed it on its latest versions, it used to have a 10s default timeout

We even have a "customAxiosInstance" class that sets this timeout to 30s to be able to load the wallet

I've added a 10s default timeout for token apis in 9034251868aa9ea5a5443a12f58e1596c3dbba87 though